code-studio/scripts/build-commands.js

```diff
@@ -252,13 +276,29 @@ exports.sass = function (srcPath, buildPath, file, includePaths, shouldMinify) {
 };
 
 /**
- * Given a filename, synchronously ensures the directory that would contain
- * that file exists (to any depth).
- * @param {string} target - path to a file.
+ * @param {!string} message
+ * @returns {Function} that logs the given message in a success style.
  */
-function ensureTargetDirectoryExists(target) {
-  mkdirp.sync(path.dirname(target));
-}
+exports.logSuccess = function (message) {
+  return function () {
+    exports.logBoxedMessage(message);
+  };
+};
+
+/**
+ * @param {!string} message
+ * @returns {Function} that logs the given message in a failure style along
+ *          with the error that triggered the failure, and rethrows the
+ *          error.
+ */
+exports.logFailure = function (message) {
+  return function (error) {
+    logBoldRedError(error);
+    warnIfWrongNodeVersion();
+    exports.logBoxedMessage(message, chalk.bold.red.bgBlack);
+    throw error;
+  };
+};
```

Comment:
Actually, I'm using lodash _.partial.  Curry doesn't work so well since, for example, `logSuccess` has all the arguments it needs before we actually want to run it.